src/fuse_api.pxi

```diff
@@ -301,9 +301,11 @@ def main(workers=None, handle_signals=True):
             on_exit.callback(lambda: restore_signal_handlers())
 
         # Start notification handling thread
+        _notify_queue_shutdown.clear()
         t = threading.Thread(target=_notify_loop)
         t.daemon = True
         t.start()
+        on_exit.callback(_notify_queue_shutdown.set)
```

Comment:
Why do we need this in addition to the sentinel object? And if the sentinel method doesn't work, shouldn't we replace it with the Event instead of doing both?